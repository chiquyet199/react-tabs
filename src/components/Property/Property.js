import React, {Fragment} from "react";
import {Linking} from "react-native";
import Dialog from "../Dialog";
import ReadMore from "../ReadMore";
import Carousel from "../Carousel";
import IconWithContent from "../IconWithContent";
import {TextLink} from "../common/layout";
import {
  BaseContainer,
  PageContainer,
  ColumnText,
  RowText,
  ContentWrapper,
  DetailsSection,
  DetailsTitle,
  DetailsText,
  DetailsSubTitle,
  MapImage,
} from "./Property.style";
import AbstractProperty from "./AbstractProperty";
// ICON IMAGES
const paymentsIcon = require("../../assets/payments.png");
const claimsIcon = require("../../assets/claims.png");
const bedroomIcon = require("../../assets/bedroom.png");
const furnishedIcon = require("../../assets/furnished.png");
const bathroomIcon = require("../../assets/bathroom.png");
const tickIcon = require("../../assets/tick.png");
const crossIcon = require("../../assets/close.png");
const mapPlaceholder = require("../../assets/map_placeholder.png");
const userIcon = require("../../assets/user.png");
const emailIcon = require("../../assets/email.png");
const phoneIcon = require("../../assets/phone.png");
const employmentIcon = require("../../assets/employment.png");

export class Property extends AbstractProperty {
  render() {
    const {
      theme,
      globaltheme,
      property,
      agency,
      branch,
      translations,
    } = this.props;
    const themeProps = {theme, globaltheme};
    const iconColor = theme && theme[globaltheme].baseColor;
    const iconsThemeProps = {theme, globaltheme, iconColor};

    return (
      <Fragment>
        <BaseContainer>
          {property && (
            <PageContainer>
              {property.propertyImages && (
                <Carousel
                  fullWidth
                  slides={property.propertyImages}
                  dots="light"
                />
              )}
              <ContentWrapper>
                <DetailsSection>
                  <DetailsTitle>{property.presentation.summary}</DetailsTitle>
                  <DetailsSubTitle>
                    {`${property.address.line1}, ${property.address.town}, ${
                      property.address.postCode
                    }`}
                  </DetailsSubTitle>
                </DetailsSection>
                <DetailsSection>
                  <DetailsTitle bold="true" mb={16}>
                    {property.rentAmount} {translations.property.pcm}
                  </DetailsTitle>

                  <IconWithContent icon={paymentsIcon} {...iconsThemeProps}>
                    <DetailsText bold="true">
                      {property.depositAmount}
                    </DetailsText>
                    <DetailsText>
                      {` ${translations.property.depositRequired}`}
                    </DetailsText>
                  </IconWithContent>

                  <IconWithContent icon={claimsIcon} {...iconsThemeProps}>
                    <ColumnText>
                      <RowText>
                        <DetailsText bold="true">
                          {translations.property.DepositFree}
                        </DetailsText>
                        <DetailsText>
                          {property.depositFreeAvailable
                            ? translations.property.available
                            : translations.property.unavailable}
                        </DetailsText>
                      </RowText>
                      <TextLink
                        gutters={0}
                        onClick={() =>
                          Linking.openURL(
                            "https://www.canopy.rent/#depositfree",
                          )
                        }
                      >
                        {`${translations.property.whatIs} ${
                          translations.property.DepositFree
                        }?`}
                      </TextLink>
                    </ColumnText>
                  </IconWithContent>
                </DetailsSection>

                <DetailsSection>
                  <IconWithContent
                    icon={bedroomIcon}
                    width={50}
                    bold="true"
                    text={`${property.presentation.bedroomsCount} ${
                      translations.property.bedrooms
                    }`}
                    {...iconsThemeProps}
                  />

                  <IconWithContent
                    icon={furnishedIcon}
                    width={50}
                    bold="true"
                    text={this.getFurnishedType(
                      property.presentation.furnishedType,
                    )}
                    {...iconsThemeProps}
                  />

                  <IconWithContent
                    icon={bathroomIcon}
                    width={50}
                    bold="true"
                    text={`${property.presentation.bedroomsCount} ${
                      translations.property.bathrooms
                    }`}
                    {...iconsThemeProps}
                  />
                </DetailsSection>

                <DetailsSection>
                  <IconWithContent
                    icon={
                      property.presentation.petsAllowed ? tickIcon : crossIcon
                    }
                    width={50}
                    bold="true"
                    text={translations.property.pets}
                    disabled={!property.presentation.petsAllowed}
                    {...iconsThemeProps}
                  />

                  <IconWithContent
                    icon={
                      property.presentation.hasParking ? tickIcon : crossIcon
                    }
                    width={50}
                    disabled={!property.presentation.hasParking}
                    bold="true"
                    text={translations.property.parking}
                    {...iconsThemeProps}
                  />

                  <IconWithContent
                    icon={
                      property.presentation.smokingAllowed
                        ? tickIcon
                        : crossIcon
                    }
                    width={50}
                    disabled={!property.presentation.smokingAllowed}
                    bold="true"
                    text={translations.property.smoking}
                    {...iconsThemeProps}
                  />

                  <IconWithContent
                    icon={
                      property.presentation.hasOutsideSpace
                        ? tickIcon
                        : crossIcon
                    }
                    width={50}
                    disabled={!property.presentation.hasOutsideSpace}
                    bold="true"
                    text={translations.property.outsideSpace}
                    {...iconsThemeProps}
                  />
                </DetailsSection>

                <DetailsSection>
                  <ReadMore
                    text={property.presentation.description}
                    {...themeProps}
                  />
                </DetailsSection>
              </ContentWrapper>

              <MapImage source={mapPlaceholder} />

              <ContentWrapper top={20}>
                <DetailsSection>
                  <ColumnText>
                    <DetailsTitle>{agency.name}</DetailsTitle>
                    {property.shareStatus === "no_rent_passport" && (
                      <DetailsText bold="true">{branch.name}</DetailsText>
                    )}
                  </ColumnText>

                  <IconWithContent icon={userIcon} {...iconsThemeProps}>
                    <DetailsTitle>
                      {`${agency.legalRepresentative.firstName} ${
                        agency.legalRepresentative.lastName
                      }`}
                    </DetailsTitle>
                  </IconWithContent>

                  {agency.phone && (
                    <IconWithContent icon={phoneIcon} {...iconsThemeProps}>
                      <TextLink
                        gutters={0}
                        onClick={() => Linking.openURL(agency.phone)}
                      >
                        {agency.phone}
                      </TextLink>
                    </IconWithContent>
                  )}

                  {agency.email && (
                    <IconWithContent icon={emailIcon} {...iconsThemeProps}>
                      <TextLink
                        gutters={0}
                        onClick={() =>
                          Linking.openURL(`mailto:${agency.email}`)
                        }
                      >
                        {agency.email}
                      </TextLink>
                    </IconWithContent>
                  )}

                  {agency.address && (
                    <IconWithContent
                      icon={employmentIcon}
                      text={`${agency.address.line1}, ${agency.address.town}, ${
                        agency.address.postCode
                      }`}
                      {...iconsThemeProps}
                    />
                  )}
                </DetailsSection>
              </ContentWrapper>
              {this.renderButton()}
              <Dialog
                {...this.dialogPropsGenerator(
                  true,
                  translations.removeDialog,
                  globaltheme,
                  agency,
                  branch,
                )}
              />

              <Dialog
                {...this.dialogPropsGenerator(
                  false,
                  translations.shareDialog,
                  globaltheme,
                  agency,
                  branch,
                )}
              />
            </PageContainer>
          )}
          {!property && (
            <DetailsSubTitle>{translations.no_data}</DetailsSubTitle>
          )}
        </BaseContainer>
      </Fragment>
    );
  }
}

export default Property;
