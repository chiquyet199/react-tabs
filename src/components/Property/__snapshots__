/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/Property/__native_snapshots__