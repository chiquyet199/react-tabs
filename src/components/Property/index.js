import PropertyDetails from "./Property";

export default PropertyDetails;
