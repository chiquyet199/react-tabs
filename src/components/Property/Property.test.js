import React from "react";
import {shallow} from "enzyme";
import {extend} from "underscore";
import Property from "./Property";
import {Button} from "../../components/Button";
import Dialog from "../../components/Dialog";
import {isWeb} from "../../utils/common";

describe("Property - page tests", () => {
  it("matches snapshot", () => {
    expect(getComponent()).toMatchSnapshot();
  });

  if (isWeb) {
    it("Renders Button and onClick on RequestRentPassport", async () => {
      const goTo = jest.fn();
      const comp = await getComponent({
        goTo,
      });
      comp.find("#Request-passport-button").simulate("click");
      expect(goTo).toHaveBeenCalled();
    });

    it("Renders RentPassportList with rentPassports", async () => {
      const rentPassports = getMockProps().sharedRentPassports;
      const comp = await getComponent();
      const rentPassportList = comp.find("#RentPassportList");
      expect(rentPassportList.props().data).toEqual(rentPassports);
    });
  }

  it("fetch property with correct id on mount", async () => {
    const {id} = getMockProps().match.params;
    const fetchPropertyById = jest.fn();
    await getComponent({
      fetchPropertyById,
    });
    expect(fetchPropertyById).toHaveBeenCalledWith(id);
  });

  it("shows expected text for page title", async () => {
    const {line1} = getMockProps().property.address;
    const showTopbarCenterText = jest.fn();
    await getComponent({
      showTopbarCenterText,
    });
    expect(showTopbarCenterText).toHaveBeenCalledWith(line1);
  });

  it("If no Rent Passport - renders Button and triggers redirect to /rent-passport", () => {
    const goTo = jest.fn();
    const mockProperty = getMockProps().property;
    mockProperty.shareStatus = "no_rent_passport";
    const property = Object.assign({}, mockProperty);
    const comp = getComponent({
      property,
      goTo,
    });
    comp.find(Button).simulate("click");
    expect(goTo).toHaveBeenCalled();
  });

  it("If Rent Passport is not shared - renders Button with proper text", () => {
    const goTo = jest.fn();
    const mockProperty = getMockProps().property;
    const buttonText = getMockProps().translations.shareRentPassport;
    mockProperty.shareStatus = "not_shared";
    const property = Object.assign({}, mockProperty);
    const comp = getComponent({
      property,
      goTo,
    });
    comp.find(Button).simulate("click");
    expect(comp.find(Button).prop("middle")).toBe(buttonText);
  });

  it("Properly calls remove property method", async () => {
    const mockProps = getMockProps();
    const deleteSharedPropertyByIds = jest.fn();
    const comp = getComponent({
      deleteSharedPropertyByIds,
    });
    const removeDialog = comp
      .find(Dialog)
      .findWhere(
        el =>
          el.prop("title") === mockProps.translations.removeDialog.dialogTitle,
      );
    const {onDialogConfirm} = removeDialog.props();
    onDialogConfirm();
    expect(deleteSharedPropertyByIds).toHaveBeenCalledWith({
      propertyId: mockProps.match.params.id,
      groupId: null,
    });
  });

  it("Properly calls share rent passport method", async () => {
    const mockProps = getMockProps();
    const sharePropertyRentPassport = jest.fn();
    const comp = getComponent({
      sharePropertyRentPassport,
    });
    const shareDialog = comp
      .find(Dialog)
      .findWhere(
        el =>
          el.prop("paragraph") ===
          mockProps.translations.shareDialog.dialogParagraph,
      );
    const {onDialogConfirm} = shareDialog.props();
    onDialogConfirm();
    expect(sharePropertyRentPassport).toHaveBeenCalledWith(
      mockProps.match.params.id,
    );
  });

  it("properly converts backend furnished type", () => {
    const comp = getComponent();
    expect(comp.instance().getFurnishedType("FULLY_FURNISHED")).toEqual(
      "Fully furnished",
    );
  });

  it("generates remove dialog with correct props", () => {
    const props = getMockProps();
    const removeDialog = getComponent(props)
      .find(Dialog)
      .at(0);
    expect(removeDialog).toHaveProp("confirmButtonType", "danger");
    expect(removeDialog).toHaveProp(
      "cancelButtonText",
      props.translations.removeDialog.dialogCancel,
    );
    expect(removeDialog).toHaveProp(
      "confirmButtonText",
      props.translations.removeDialog.dialogConfirm,
    );
    expect(removeDialog).toHaveProp("globaltheme", props.globaltheme);
    expect(removeDialog).toHaveProp(
      "title",
      props.translations.removeDialog.dialogTitle,
    );
    expect(removeDialog).toHaveProp(
      "paragraph",
      `${props.translations.removeDialog.dialogParagraph1} "${
        props.agency.name
      }-${props.branch.name}"? ${
        props.translations.removeDialog.dialogParagraph2
      }`,
    );
  });

  it("generates share dialog with correct props", () => {
    const props = getMockProps();
    const shareDialog = getComponent(props)
      .find(Dialog)
      .at(1);
    expect(shareDialog).toHaveProp("confirmButtonType", "danger");
    expect(shareDialog).toHaveProp(
      "cancelButtonText",
      props.translations.shareDialog.dialogCancel,
    );
    expect(shareDialog).toHaveProp(
      "confirmButtonText",
      props.translations.shareDialog.dialogConfirm,
    );
    expect(shareDialog).toHaveProp("globaltheme", props.globaltheme);
    expect(shareDialog).toHaveProp(
      "title",
      `${props.translations.shareDialog.dialogTitle} "${props.agency.name},${
        props.branch.name
      }"?`,
    );
    expect(shareDialog).toHaveProp(
      "paragraph",
      props.translations.shareDialog.dialogParagraph,
    );
  });

  const getComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<Property {...parsedProps} />);
  };

  const getMockProps = () => ({
    match: {params: {id: "12"}},
    globaltheme: "light",
    fetchSharedPassportsWithProperty: jest.fn(),
    resetCheckedPassport: jest.fn(),
    passportsSelected: [],
    sharedRentPassports: defaultPassports,
    addPassportsToShare: () => {},
    selectedPassportType: "shared",
    isAgent: true,
    theme: {
      light: {
        baseColor: "#6b7c93",
        lightSteel: "#e3eaf2",
        veryLightSteel: "#f6f9fc",
      },
    },
    property: {
      presentation: {
        propertyType: "APARTMENT",
        bedroomsCount: 2,
        bathroomsCount: 2,
        furnishedType: "PART_FURNISHED",
        hasParking: true,
        hasOutsideSpace: true,
        petsAllowed: true,
        smokingAllowed: false,
        dateAvailable: "11-12-2019",
        summary: "2 bed apartment to rent",
        description:
          "Well presented ground floor apartment available on Stamford Street. The property is offered unfurnished and comprises entrance hallway, downstairs WC, lounge/diner with doors to balcony/terrace, storage cupboards, kitchen with white goods including electric cooke. Well presented ground floor apartment available on Stamford Street. The property is offered unfurnished and comprises entrance hallway, downstairs WC, lounge/diner with doors to balcony/terrace, storage cupboards, kitchen with white goods including electric cooker.",
      },
      rentFrequency: "MONTHLY",
      rentAmount: 1200,
      depositAmount: 1800,
      depositFreeAvailable: true,
      landlord: {
        firstName: "firstName",
        lastName: "lastName",
        email: "landlord1@gmail.com",
        phone: "01234567890",
        address: {
          line1: "Address line 1",
          line2: "Address line 2",
          town: "City / Town",
          state: "State",
          postCode: "FYC123",
          countryCode: "UK",
        },
      },
      createdBy: "67e4b887-466e-4abe-8d12-bc87204e5df8",
      agencyId: "67e4b887-466e-4abe-8d12-bc87204e5df8",
      branchId: "01a5b187-313s-7gtc-4a31-av51034r1ea1",
      createdAt: "01-01-2017",
      updatedAt: "01-01-2017",
      propertyImages: [
        {
          url:
            "http://cdn.home-designing.com/wp-content/uploads/2016/04/luxury-art-deco-apartment-interior.jpg",
          isPrimary: true,
          createdAt: "01-01-2017",
          updatedAt: "01-01-2017",
          version: 1,
        },
        {
          url:
            "http://cdn.home-designing.com/wp-content/uploads/2016/04/luxury-art-deco-apartment-interior.jpg",
          isPrimary: false,
          createdAt: "01-01-2017",
          updatedAt: "01-01-2017",
          version: 1,
        },
        {
          url:
            "http://cdn.home-designing.com/wp-content/uploads/2016/04/luxury-art-deco-apartment-interior.jpg",
          isPrimary: false,
          createdAt: "01-01-2017",
          updatedAt: "01-01-2017",
          version: 1,
        },
      ],
      address: {
        line1: "Address line 1",
        line2: "Address line 2",
        town: "City / Town",
        state: "State",
        postCode: "FYC123",
        countryCode: "UK",
      },
      shareStatus: "not_shared",
    },
    agency: {
      id: "a71bbc1c-3fe5-4c2f-8083-d4e1055f2f18",
      name: "Real Agency",
      type: "ESTATE_AGENCY",
      companyRegistrationNumber: "01234567",
      phone: "0123456790",
      address: {
        line1: "Address line 1",
        line2: "Address line 2",
        town: "City / Town",
        state: "State",
        postCode: "FYC123",
        countryCode: "UK",
      },
      agents: [],
      legalRepresentative: {
        firstName: "Becky",
        lastName: "Hammond",
        address: {
          line1: "Address line 1",
          line2: "Address line 2",
          town: "City / Town",
          state: "State",
          postCode: "FYC123",
          countryCode: "UK",
        },
        dob: "01-01-1980",
      },
      bankAccount: {
        accountHolder: "accountHolder",
        accountNumber: "accountNumber",
        sortCode: "sortCode",
      },
      isActive: true,
    },
    branch: {
      id: "1bc9925d-6ae6-4448-9523-05912cdce917",
      agencyId: "c9a463b0-b019-45b6-be57-720eb28d0bc0",
      name: "Chelsea",
      address: {
        line1: "Address line 1",
        line2: "Address line 2",
        town: "City / Town",
        state: "State",
        postCode: "FYC123",
        countryCode: "UK",
      },
      phone: "01123456789",
    },
    translations: {
      removePassportDialog: {
        title: "Delete Passport",
        pendingTitle: "Delete Request",
        paragraph: "Are you sure you want to delete the selected passport?",
        pendingParagraph: "Are you sure you want to delete the request?",
        cancelButtonText: "Cancel",
        confirmButtonText: "Confirm",
      },
      removeDialog: {
        dialogTitle: "Confirm stop sharing property",
        dialogParagraph1: "Are you sure you want to stop sharing with",
        dialogParagraph2: "This stop sharing with all associated properties",
        dialogCancel: "Cancel",
        dialogConfirm: "Stop sharing",
      },
      shareDialog: {
        dialogTitle: "Share your Rent Passport with",
        dialogParagraph:
          "This will share all information in your Rent Passport and qualify you against the property.",
        dialogCancel: "Cancel",
        dialogConfirm: "Share",
      },
      property: {
        pcm: "pcm",
        depositRequired: "deposit required",
        DepositFree: "Deposit Free",
        available: "available",
        unavailable: "unavailable",
        whatIs: "What is",
        bedrooms: "bedrooms",
        bathrooms: "bathrooms",
        pets: "Pets",
        parking: "Parking",
        smoking: "Smoking",
        outsideSpace: "Outside space",
      },
      shareRentPassport: "Share Rent Passport",
      createRentPassport: "Create a RentPassport to rent this property.",
      whatIsRentPassport: "What is a Rent Passport?",
      startRentPassport: "Start my Rent Passport",
    },
    fetchPropertyById: () => {},
    deleteSharedPropertyByIds: () => {},
    sharePropertyRentPassport: () => {},
    updateTopbarButton: () => {},
    showTopbarButton: () => {},
    showTopbarCenterText: () => {},
  });
});

const defaultPassports = [
  {
    id: 777,
    name: "Jonathan Beaumont",
    status: "Fully referenced",
  },
  {
    id: 666,
    name: "James Wang",
    status: "Partially referenced",
  },
  {
    id: 111,
    name: "Jennifer Jones",
    status: "Not fully referenced",
  },
  {
    id: 133,
    name: "joshua_yates99@hotmail.com",
    pending: true,
  },
  {
    id: 134,
    name: "bob_marley66@hotmail.com",
    pending: true,
  },
];
