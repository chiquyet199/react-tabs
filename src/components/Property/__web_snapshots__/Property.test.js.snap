// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property - page tests matches snapshot 1`] = `
<Styled(View)>
  <Styled(Component)>
    <Connect(ResponsiveWrapper)>
      <View
        style={
          Object {
            "flex": 2,
            "padding": 18,
          }
        }
      >
        <Connect(WithTheme(Styled(View)))>
          <Connect(WithTheme(Styled(Text)))>
            2 bed apartment to rent
          </Connect(WithTheme(Styled(Text)))>
          <Connect(WithTheme(Styled(Text)))>
            Address line 1, City / Town, FYC123
          </Connect(WithTheme(Styled(Text)))>
        </Connect(WithTheme(Styled(View)))>
        <Connect(WithTheme(Styled(View)))>
          <Connect(WithTheme(Styled(Text)))
            bold="true"
            mb={16}
          >
            1200
             
            pcm
          </Connect(WithTheme(Styled(Text)))>
          <IconWithContent
            globaltheme="light"
            iconColor="#6b7c93"
            iconName="payments"
            theme={
              Object {
                "light": Object {
                  "baseColor": "#6b7c93",
                  "lightSteel": "#e3eaf2",
                  "veryLightSteel": "#f6f9fc",
                },
              }
            }
          >
            <Connect(WithTheme(Styled(Text)))>
              <strong>
                1800
              </strong>
               deposit required
            </Connect(WithTheme(Styled(Text)))>
          </IconWithContent>
          <IconWithContent
            globaltheme="light"
            iconColor="#6b7c93"
            iconName="claims"
            theme={
              Object {
                "light": Object {
                  "baseColor": "#6b7c93",
                  "lightSteel": "#e3eaf2",
                  "veryLightSteel": "#f6f9fc",
                },
              }
            }
          >
            <Styled(View)>
              <Connect(WithTheme(Styled(Text)))>
                <strong>
                  Deposit Free
                   
                </strong>
                available
              </Connect(WithTheme(Styled(Text)))>
              <TextLink
                gutters={0}
                onClick={[Function]}
              >
                What is Deposit Free?
              </TextLink>
            </Styled(View)>
          </IconWithContent>
          <Styled(View)>
            <Connect(WithTheme(Button))
              id="Request-passport-button"
              onClick={[Function]}
              type="secondary"
            />
          </Styled(View)>
        </Connect(WithTheme(Styled(View)))>
        <Connect(WithTheme(Styled(View)))>
          <RentPassportList
            addPassportsToShare={[Function]}
            data={
              Array [
                Object {
                  "id": 777,
                  "name": "Jonathan Beaumont",
                  "status": "Fully referenced",
                },
                Object {
                  "id": 666,
                  "name": "James Wang",
                  "status": "Partially referenced",
                },
                Object {
                  "id": 111,
                  "name": "Jennifer Jones",
                  "status": "Not fully referenced",
                },
                Object {
                  "id": 133,
                  "name": "joshua_yates99@hotmail.com",
                  "pending": true,
                },
                Object {
                  "id": 134,
                  "name": "bob_marley66@hotmail.com",
                  "pending": true,
                },
              ]
            }
            globaltheme="light"
            id="RentPassportList"
            selectedPassport="shared"
            theme={
              Object {
                "light": Object {
                  "baseColor": "#6b7c93",
                  "lightSteel": "#e3eaf2",
                  "veryLightSteel": "#f6f9fc",
                },
              }
            }
          />
        </Connect(WithTheme(Styled(View)))>
      </View>
      <View
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <Connect(WithTheme(WebCarousel))
          dots="light"
          fullWidth={true}
          slides={
            Array [
              Object {
                "createdAt": "01-01-2017",
                "isPrimary": true,
                "updatedAt": "01-01-2017",
                "url": "http://cdn.home-designing.com/wp-content/uploads/2016/04/luxury-art-deco-apartment-interior.jpg",
                "version": 1,
              },
              Object {
                "createdAt": "01-01-2017",
                "isPrimary": false,
                "updatedAt": "01-01-2017",
                "url": "http://cdn.home-designing.com/wp-content/uploads/2016/04/luxury-art-deco-apartment-interior.jpg",
                "version": 1,
              },
              Object {
                "createdAt": "01-01-2017",
                "isPrimary": false,
                "updatedAt": "01-01-2017",
                "url": "http://cdn.home-designing.com/wp-content/uploads/2016/04/luxury-art-deco-apartment-interior.jpg",
                "version": 1,
              },
            ]
          }
          wrapAround={false}
        />
        <Styled(View)>
          <Connect(WithTheme(Styled(View)))>
            <IconWithContent
              bold="true"
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="bedroom"
              text="2 bedrooms"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
              width={50}
            />
            <IconWithContent
              bold="true"
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="furnished"
              text="Part furnished"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
              width={50}
            />
            <IconWithContent
              bold="true"
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="bathroom"
              text="2 bathrooms"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
              width={50}
            />
          </Connect(WithTheme(Styled(View)))>
          <Connect(WithTheme(Styled(View)))>
            <IconWithContent
              bold="true"
              disabled={false}
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="tick"
              text="Pets"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
              width={50}
            />
            <IconWithContent
              bold="true"
              disabled={false}
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="tick"
              text="Parking"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
              width={50}
            />
            <IconWithContent
              bold="true"
              disabled={true}
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="close"
              text="Smoking"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
              width={50}
            />
            <IconWithContent
              bold="true"
              disabled={false}
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="tick"
              text="Outside space"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
              width={50}
            />
          </Connect(WithTheme(Styled(View)))>
          <Connect(WithTheme(Styled(View)))>
            <Connect(WithTheme(ReadMoreWeb))
              globaltheme="light"
              text="Well presented ground floor apartment available on Stamford Street. The property is offered unfurnished and comprises entrance hallway, downstairs WC, lounge/diner with doors to balcony/terrace, storage cupboards, kitchen with white goods including electric cooke. Well presented ground floor apartment available on Stamford Street. The property is offered unfurnished and comprises entrance hallway, downstairs WC, lounge/diner with doors to balcony/terrace, storage cupboards, kitchen with white goods including electric cooker."
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
            />
          </Connect(WithTheme(Styled(View)))>
        </Styled(View)>
        <Connect(ResponsiveWrapper)
          source="map_placeholder.png"
        />
        <Styled(View)
          top={20}
        >
          <Connect(WithTheme(Styled(View)))>
            <Styled(View)>
              <Connect(WithTheme(Styled(Text)))>
                Real Agency
              </Connect(WithTheme(Styled(Text)))>
            </Styled(View)>
            <IconWithContent
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="user"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
            >
              <Connect(WithTheme(Styled(Text)))>
                Becky Hammond
              </Connect(WithTheme(Styled(Text)))>
            </IconWithContent>
            <IconWithContent
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="phone"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
            >
              <TextLink
                gutters={0}
                onClick={[Function]}
              >
                0123456790
              </TextLink>
            </IconWithContent>
            <IconWithContent
              globaltheme="light"
              iconColor="#6b7c93"
              iconName="employment"
              text="Address line 1, City / Town, FYC123"
              theme={
                Object {
                  "light": Object {
                    "baseColor": "#6b7c93",
                    "lightSteel": "#e3eaf2",
                    "veryLightSteel": "#f6f9fc",
                  },
                }
              }
            />
          </Connect(WithTheme(Styled(View)))>
        </Styled(View)>
        <Styled(View)>
          <Button
            globaltheme="light"
            iconsize={32}
            left={null}
            middle="Share Rent Passport"
            onClick={[Function]}
            right="arrow-right"
            theme={
              Object {
                "light": Object {
                  "baseColor": "#6b7c93",
                  "lightSteel": "#e3eaf2",
                  "veryLightSteel": "#f6f9fc",
                },
              }
            }
            type="primary"
          />
        </Styled(View)>
        <Connect(WithTheme(SimpleDialog))
          cancelButtonText="Cancel"
          confirmButtonText="Stop sharing"
          confirmButtonType="danger"
          globaltheme="light"
          isOpen={false}
          onClose={[Function]}
          onDialogCancel={[Function]}
          onDialogConfirm={[Function]}
          paragraph="Are you sure you want to stop sharing with \\"Real Agency-Chelsea\\"? This stop sharing with all associated properties"
          title="Confirm stop sharing property"
        />
        <Connect(WithTheme(SimpleDialog))
          cancelButtonText="Cancel"
          confirmButtonText="Share"
          confirmButtonType="danger"
          globaltheme="light"
          isOpen={false}
          onClose={[Function]}
          onDialogCancel={[Function]}
          onDialogConfirm={[Function]}
          paragraph="This will share all information in your Rent Passport and qualify you against the property."
          title="Share your Rent Passport with \\"Real Agency,Chelsea\\"?"
        />
        <Connect(WithTheme(SimpleDialog))
          cancelButtonText="Cancel"
          confirmButtonText="Confirm"
          globaltheme="light"
          isOpen={false}
          onDialogCancel={[Function]}
          onDialogConfirm={[Function]}
          paragraph="Are you sure you want to delete the selected passport?"
          theme={
            Object {
              "light": Object {
                "baseColor": "#6b7c93",
                "lightSteel": "#e3eaf2",
                "veryLightSteel": "#f6f9fc",
              },
            }
          }
          title="Delete Passport"
        />
      </View>
    </Connect(ResponsiveWrapper)>
  </Styled(Component)>
</Styled(View)>
`;
